src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -442,6 +446,25 @@ def generate_config(
         prometheus_user, prometheus_password = self.mgr._get_prometheus_credentials()
         FSID = self.mgr._cluster_fsid
 
+        clusters_credentials = {}
+        multi_cluster_config_str = str(self.mgr.get_module_option_ex('dashboard', 'MULTICLUSTER_CONFIG'))
+        try:
+            multi_cluster_config = literal_eval(multi_cluster_config_str)  # convert python string to dict
+        except ValueError:
+            multi_cluster_config = None
+            logger.error("Invalid JSON format for multi-cluster config")
+        try:
+            if multi_cluster_config:
+                for url in targets:
+                    # Finding Prometheus credentials for each URL
+                    credentials = self.find_prometheus_credentials(multi_cluster_config, url)
+                    if credentials:
+                        # Storing credentials in the dictionary
```

Comment:
same here .. in general comments should explain some details that are not obvious and it's not the case